test/helpers/framework.go

```diff
@@ -139,13 +196,38 @@ func (tc *E2ETestFramework) WaitFor(component LogComponentType) error {
 		return tc.waitForDeployment(OpenshiftLoggingNS, "kibana", defaultRetryInterval, defaultTimeout)
 	case ComponentTypeCollector:
 		logger.Debugf("Waiting for %v", component)
-		return e2eutil.WaitForDaemonSet(&testing.T{}, tc.KubeClient, OpenshiftLoggingNS, "fluentd", defaultRetryInterval, defaultTimeout)
+		return tc.waitForFluentDaemonSet(defaultRetryInterval, time.Minute*5)
 	case ComponentTypeStore:
 		return tc.waitForElasticsearchPods(defaultRetryInterval, defaultTimeout)
 	}
 	return fmt.Errorf("Unable to waitfor unrecognized component: %v", component)
 }
 
+func (tc *E2ETestFramework) waitForFluentDaemonSet(retryInterval, timeout time.Duration) error {
+	// daemonset should have non-zero number of instances for maxtimes consecutive retryInterval to detect a CrashLoopBackOff pod
+	maxtimes := 5
+	times := 0
+	return wait.Poll(retryInterval, timeout, func() (bool, error) {
+		numReady, err := oc.Literal().From("oc -n openshift-logging get daemonset/fluentd -o jsonpath={.status.numberReady}").Run()
```

Comment:
Consider using the kubeclient here as other wait functions in our test framework code. Re-using one pattern is easier to maintain, e.g. `tc.KubeClient.AppsV1().Daemonset(..)`

Reply:
for querying daemonset there is no difference between oc and kubectl. Also the literal command begins with `oc` because it is an `ochelper`

Reply:
Sure there is no difference in functionality terms, but we have multiple wait methods written using the kubeclient. Would be nice to keep the pattern here.

Reply:
the expression is querying the daemonset, and is written using `oc.helper` , a kubectl will look rather odd here